File: medplum-typescript/packages/sdk/src/model/fhir/CoverageEligibilityRequest.ts
import { code } from "./code";
import { CodeableConcept } from "./CodeableConcept";
import { CoverageeligibilityrequestPurposeItem } from "./CoverageeligibilityrequestPurposeItem";
import { CoverageEligibilityRequest_Insurance } from "./CoverageEligibilityRequest_Insurance";
import { CoverageEligibilityRequest_Item } from "./CoverageEligibilityRequest_Item";
import { CoverageEligibilityRequest_SupportingInfo } from "./CoverageEligibilityRequest_SupportingInfo";
import { dateTime } from "./dateTime";
import { Extension } from "./Extension";
import { id } from "./id";
import { Identifier } from "./Identifier";
import { Meta } from "./Meta";
import { Narrative } from "./Narrative";
import { Period } from "./Period";
import { Reference } from "./Reference";
import { ResourceList } from "./ResourceList";
import { uri } from "./uri";

export interface CoverageEligibilityRequest {
    resourceType: string;
    id: id | null | undefined;
    meta: Meta | null | undefined;
    implicitRules: uri | null | undefined;
    language: code | null | undefined;
    text: Narrative | null | undefined;
    contained: ResourceList[] | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    identifier: Identifier[] | null | undefined;
    status: code | null | undefined;
    priority: CodeableConcept | null | undefined;
    purpose: CoverageeligibilityrequestPurposeItem[] | null | undefined;
    patient: Reference;
    servicedDate: string | null | undefined;
    servicedPeriod: Period | null | undefined;
    created: dateTime | null | undefined;
    enterer: Reference | null | undefined;
    provider: Reference | null | undefined;
    insurer: Reference;
    facility: Reference | null | undefined;
    supportingInfo:
        | CoverageEligibilityRequest_SupportingInfo[]
        | null
        | undefined;
    insurance: CoverageEligibilityRequest_Insurance[] | null | undefined;
    item: CoverageEligibilityRequest_Item[] | null | undefined;
}

File: medplum-java/medplum-java-model/src/main/java/com/fhir/CoverageEligibilityRequest.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableCoverageEligibilityRequest.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface CoverageEligibilityRequest {
  Optional<CodeableConcept> priority();

  Optional<String> servicedDate();

  Optional<List<Identifier>> identifier();

  Optional<id> id();

  Optional<List<CoverageeligibilityrequestPurposeItem>> purpose();

  Optional<Narrative> text();

  Optional<List<CoverageEligibilityRequest_SupportingInfo>> supportingInfo();

  Optional<code> status();

  Optional<uri> implicitRules();

  Optional<List<CoverageEligibilityRequest_Insurance>> insurance();

  Optional<List<CoverageEligibilityRequest_Item>> item();

  Optional<Reference> enterer();

  Optional<Reference> provider();

  Optional<code> language();

  Reference insurer();

  Optional<Period> servicedPeriod();

  Optional<List<ResourceList>> contained();

  Optional<List<Extension>> modifierExtension();

  Optional<List<Extension>> extension();

  Optional<Meta> meta();

  String resourceType();

  Optional<dateTime> created();

  Optional<Reference> facility();

  Reference patient();

  static ImmutableCoverageEligibilityRequest.InsurerBuildStage builder() {
    return ImmutableCoverageEligibilityRequest.builder();
  }
}
